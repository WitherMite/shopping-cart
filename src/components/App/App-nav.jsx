import { Link } from "react-router-dom";

export default function AppNav() {
  return (
    <nav>
      <Link>Home</Link>
      <Link>Store</Link>
      <Link>Cart</Link>
    </nav>
  );
}
