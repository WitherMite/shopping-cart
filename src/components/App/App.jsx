import { Outlet } from "react-router-dom";
import { useState } from "react";
import PropTypes from "prop-types";
import AppNav from "./App-nav";

export default function App({ navBar }) {
  const [cart, setCart] = useState([]);
  const navigation = navBar || <AppNav cart={cart} setCart={setCart}></AppNav>;
  return (
    <>
      <header>
        <img src="" alt="" />
        {navigation}
      </header>
      <main>
        <Outlet context={[cart, setCart]} />
      </main>
    </>
  );
}

App.propTypes = {
  navBar: PropTypes.element,
};
