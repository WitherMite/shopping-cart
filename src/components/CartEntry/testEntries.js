import logo from "../../assets/logo-ipsum-small.svg";

const testEntries = [
  {
    id: 1,
    title: "Item 1",
    price: "40",
    category: "test",
    description:
      "This is item number 1, there are many like it, but this is the first",
    image: logo,
    count: 1,
  },
  {
    id: 2,
    title: "Item 2",
    price: "12",
    category: "test",
    description: "Lorem ipsum sit amet dolor...",
    image: logo,
    count: 50,
  },
  {
    id: 3,
    title: "Item 3",
    price: "20.45",
    category: "test",
    description: "Three.",
    image: logo,
    count: 3,
  },
];

export default testEntries;
