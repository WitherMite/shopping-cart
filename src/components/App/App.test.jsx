import { RouterProvider, createMemoryRouter } from "react-router-dom";
import { describe, it, expect } from "vitest";
import { render, screen, within } from "@testing-library/react";
import testEntries from "../CartEntry/testEntries.js";
import App from "./App.jsx";

describe("App component", () => {
  it("Renders a logo and nav bar inside a header and a main tag", () => {
    const router = createMemoryRouter([{ path: "/", element: <App /> }]);
    const { container } = render(<RouterProvider router={router} />);
    expect(container).toMatchSnapshot();
  });

  describe("Nav bar", () => {
    it("cart icon has a notification showing item count", () => {
      const router = createMemoryRouter([
        { path: "/", element: <App initialCart={testEntries} /> },
      ]);
      render(<RouterProvider router={router} />);

      const itemCount = testEntries.reduce(
        (count, entry) => count + entry.count,
        0
      );
      const cartIcon = screen.getByRole("link", { name: "Cart" });
      expect(() => within(cartIcon).getByText(`${itemCount}`)).not.toThrow();
    });
    it("cart icon does not show a notification for empty cart", () => {
      const router = createMemoryRouter([{ path: "/", element: <App /> }]);
      render(<RouterProvider router={router} />);

      const cartIcon = screen.getByRole("link", { name: "Cart" });
      expect(() => within(cartIcon).getByText("0")).toThrow();
    });
  });
});
