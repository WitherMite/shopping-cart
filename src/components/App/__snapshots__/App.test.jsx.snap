// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App component > Renders a logo and nav bar inside a header and a main tag 1`] = `
<div>
  <header>
    <div
      class="header-flex-container"
    >
      <a
        href="/"
      >
        <img
          alt=""
          src="/src/assets/logoipsum-265.svg"
        />
      </a>
      <nav>
        <a
          href="/"
        >
          <img
            alt="Home"
            class="nav-icon"
            src="/src/assets/home.svg"
          />
        </a>
        <a
          href="/store"
        >
          <img
            alt="Store"
            class="nav-icon"
            src="/src/assets/shopping-bag.svg"
          />
        </a>
        <a
          aria-labelledby="cartLinkImg"
          href="/cart"
        >
          <img
            alt="Cart"
            class="nav-icon"
            id="cartLinkImg"
            src="/src/assets/shopping-cart.svg"
          />
        </a>
      </nav>
    </div>
  </header>
  <main />
</div>
`;
